### PR TITLE
hammer: osd: do not send ENXIO on misdirected op by default

### PR DESCRIPTION
**Pull request by shinobu-x**
http://tracker.ceph.com/issues/18813

### CODE REVIEW

File: PendingReleaseNotes [hunk 1]
```diff
@@ -18,3 +18,32 @@ v0.94.3
...
+   the standalone calls in the C++ interface
+   (IoCtx::get_omap_{keys,vals}), librados has been updated to loop on
+   the client side to provide a correct result via multiple calls to
+   the OSD.  In the case of the methods used for building
+   multi-operation transactions, however, client-side looping is not
+   practical, and the methods have been deprecated.  Note that use of
+   either the IoCtx methods on older librados versions or the
+   deprecated methods on any version of librados will lead to
+   incomplete results if/when the new OSD limits are enabled.
+
```
Comment (liewegas) [inline PendingReleaseNotes]: the above shouldn't be included

Comment (shinobu-x) [inline PendingReleaseNotes]: Got it! Sorry for that :)

**Comment by wido** [comment]
But this is a behavior change, right. We are adding new settings and change how OSDs react.

Do we want to do this in a point release of a LTS?

**Comment by liewegas** [comment]
I think so.  ENXIO should only occur if there is a bug in ceph.  This just 
changes the failure mode to be a stalled IO (with possible 
recovery) instead of EIO (and read-only VM file system).

**Comment by smithfarm** [comment]
@liewegas Why would we not include the release notes entry in hammer?

File: PendingReleaseNotes (hunk 1 above)
Comment (smithfarm) [inline PendingReleaseNotes]: @liewegas Why would we not include the release notes entry in hammer?

**Comment by liewegas** [comment]
i just meant the note for the item tht wasn't included in teh pr.. it 
looked like it was dragged in from a merge conflict

**Comment by smithfarm** [comment]
@shinobu-x Please include the PendingReleaseNotes item from https://github.com/ceph/ceph/commit/923e7f5ce5ed437af15e178299a61029ff48e4a2 - don't change the existing content; just add a `0.94.11` heading above the new entry.

**Comment by shinobu-x** [comment]
@smithfarm Repushed with suggested file including a new line. Please check.

**Comment by smithfarm** [comment]
@shinobu-x Great, now delete lines 4-34 and you'll have it right :-)

File: PendingReleaseNotes [hunk 2]
```diff
@@ -1,20 +0,0 @@
...
-* ceph-objectstore-tool and ceph-monstore-tool now enable user to
-  rebuild the monitor database from OSDs. This feature is especially
-  useful when all monitors fail to boot due to leveldb corruption.
-
-v0.94.3
--------
-
-* The return code for librbd's rbd_aio_read and Image::aio_read API methods no
-  longer returns the number of bytes read upon success.  Instead, it returns 0
-  upon success and a negative value upon failure.
```
Comment (smithfarm) [inline PendingReleaseNotes]: This stuff is obsolete so it's OK to drop it.

**Comment by smithfarm** [comment]
Hammer is EOL.

**Pull request closed without merging**